concent_api/core/models.py

```diff
@@ -494,70 +494,53 @@ def clean(self) -> None:
             raise ValidationError({
                 'computation_deadline': "TaskToCompute deadline mismatch"
             })
+
         # Validation for every nested message which is stored in Control database
         # Every nested message must be the same as message stored separately.
-        if deserialized_report_computed_task.task_to_compute != deserialized_task_to_compute:
-            raise ValidationError({
-                'report_computed_task': 'Nested ReportComputedTask with TaskToCompute message '
-                                        'must be the same as both messages stored separately'
-            })
-        if self.ack_report_computed_task is not None:
-            if (
-                deserialize_message(self.ack_report_computed_task.data).report_computed_task !=
-                deserialized_report_computed_task
-            ):
-                raise ValidationError({
-                    'ack_report_computed_task': 'Nested ReportComputedTask with TaskToCompute message '
-                                                'must be the same as both messages stored separately'
-                })
-        if self.reject_report_computed_task is not None:
-            deserialized_reject_report_computed_task = deserialize_message(self.reject_report_computed_task.data)
-            if deserialized_reject_report_computed_task.reason == message.tasks.RejectReportComputedTask.REASON.GotMessageCannotComputeTask.name:  # pylint: disable=no-member
-                if deserialized_reject_report_computed_task.cannot_compute_task.task_to_compute != deserialized_task_to_compute:
+        def validate_consistence_of_nested_messages(message_1: message.Message, message_2: Optional[StoredMessage]=None) -> None:
+            if message_2 is None:
+                return
+            if isinstance(message_2.data, bytes):
```

Comment:
This is impossible to get anything else. If message will be None -> we will return from function. After this we have just 2 possibilities of data - it could be `bytes` or `memoryview` because it depends on when this validation starts. If this starts when `Subtask` model is created then these messages could by `None` or `bytes`. If new messages are stored and assigned to this `Subtask` then new messages will be `bytes`, but old ones, which are already stored in database are `memoryview` type and then we have to pass them in different way to `deserialize_message` function.